cdap-app-fabric/src/main/java/io/cdap/cdap/security/executor/SecurityContextRunnables.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.executor;
+
+import io.cdap.cdap.proto.security.Credential;
+import io.cdap.cdap.security.spi.authentication.SecurityRequestContext;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Class containing utility methods for executor thread pools.
+ */
+public class SecurityContextRunnables {
+  static Runnable wrapRunnableWithContext(Runnable runnable) {
+    String userID = SecurityRequestContext.getUserId();
+    Credential userCredential = SecurityRequestContext.getUserCredential();
+    String userIP = SecurityRequestContext.getUserIP();
+    return () -> {
+      SecurityRequestContext.setUserId(userID);
+      SecurityRequestContext.setUserCredential(userCredential);
+      SecurityRequestContext.setUserIP(userIP);
+      try {
+        runnable.run();
+      } finally {
+        SecurityRequestContext.reset();
```

Comment:
if there was a security context already set, this will wipe it out. I think this should be fine from the TwillRunnerService point of view, but it might be better to set these back to whatever value they were before in case they were non-null?

Reply:
they are used that way now, but this is just a utility class that can be called. Maybe move these things into the ContextInheritingExecutor then?